pkg/executor/statefulset.go

```diff
@@ -273,18 +274,35 @@ func getStatefulset(
 		return nil, securityErr
 	}
 
-	return &appsv1.StatefulSet{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "StatefulSet",
-			APIVersion: "apps/v1",
-		},
-		ObjectMeta: metav1.ObjectMeta{
-			GenerateName:    statefulSetNamePrefix,
+	namingScheme := *cr.Spec.NamingScheme
+	var objectmeta metav1.ObjectMeta
+	var objectName string
+	if namingScheme == v1beta1.CrNameRole {
+		objectName = ValidateName(cr.Name + "-" + role.Name)
```

Comment:
Similar comment here about uniqueness. Since ValidateName (probably should be called something else) will munge the name, it could result in the same name as some other existing statefulset. Actually I guess that's technically true even if we were not munging the name.

I know I had previously asked why there was any need to still use GenerateName in this scheme, so I guess that's the answer as to why.

Reply:
There's also the outside possibility that we would could munge two roles in the same kdcluster to have the same statefulset name. Assuming that we're using GenerateName then they will at least get names with different UID suffixes, so things will work, but it could be confusing.

However I don't think that's worth worrying much about. If someone e.g. names two roles in their kdapp "my.role" and "my-role" then they probably deserve whatever confusion results.